pkg/obs/controllerserver.go

```diff
@@ -29,7 +32,51 @@ type controllerServer struct {
 
 func (cs *controllerServer) CreateVolume(_ context.Context, req *csi.CreateVolumeRequest) (
 	*csi.CreateVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	log.Infof("CreateVolume: called with args %v", protosanitizer.StripSecrets(*req))
+	credentials := cs.Driver.cloud
+
+	volName := req.GetName()
+	if err := createVolumeValidation(volName, req.GetVolumeCapabilities()); err != nil {
+		return nil, err
+	}
+
+	if vol, err := services.GetBucket(credentials, volName); err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	} else if vol != "" {
+		return buildCreateVolumeResponse(volName), nil
+	}
+
+	_, err := services.CreateBucketWithTag(credentials, volName)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = services.GetBucket(credentials, volName)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("Successfully created volume %s", volName)
+	return buildCreateVolumeResponse(volName), nil
+}
+
+func createVolumeValidation(volumeName string, capabilities []*csi.VolumeCapability) error {
+	if len(volumeName) == 0 {
+		return status.Error(codes.InvalidArgument, "Validation failed, volume name cannot be empty")
+	}
+	if len(capabilities) == 0 {
+		return status.Error(codes.InvalidArgument, "Validation failed, volume capabilities cannot be empty")
+	}
+	return nil
+}
+
+func buildCreateVolumeResponse(volName string) *csi.CreateVolumeResponse {
+	response := &csi.CreateVolumeResponse{
+		Volume: &csi.Volume{
+			VolumeId: volName,
+		},
+	}
+	return response
```

Comment:
how about this:
return  &csi.CreateVolumeResponse{
		Volume: &csi.Volume{
			VolumeId: volName,
		},
	}